src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1563,5 +1564,80 @@ public async Task Http2GetAsync_TrailigPseudo_Throw()
                 await Assert.ThrowsAsync<HttpRequestException>(() => sendTask);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(MaxResponseHeadersLength_ExactData))]
+        public async Task MaxResponseHeadersLength_Exact_Success(string _, byte[] headerData)
```

Comment:
There are some existing tests for MaxResponseHeadersLength. did you check if it would be practical to merge them and use GenericLoopback? 
That is approach we took for some other tests - like Cookies.

Reply:
Yea I saw them; I don't see a reasonable way to do this as both HTTP/1 and HTTP/2 tests rely on carefully crafted protocol-specific data.